primitive-types/impls/serde/src/serialize.rs

```diff
@@ -36,6 +61,48 @@ fn to_hex<'a>(v: &'a mut [u8], bytes: &[u8], skip_leading_zero: bool) -> &'a str
 	unsafe { std::str::from_utf8_unchecked(&v[0..idx]) }
 }
 
+/// Decode given hex string into a vector of bytes.
+///
+/// Returns an error if the string is not prefixed with `0x`
+/// or non-hex characters are present.
+pub fn from_hex(v: &str) -> Result<Vec<u8>, String> {
```

Comment:
I thought you were against `String` error types :P

Reply:
Yeah, you're right. I just simply extracted the method, but haven't thought of proper error type here. Let me do a follow up PR on this :)